src/service/video-manager-impl.js

```diff
@@ -185,11 +184,32 @@ export class VideoManager {
     if (!video.supportsPlatform()) {
       return;
     }
+    if (!this.viewportObserver_) {
+      const viewportCallback = (
+        /** @type {!Array<!IntersectionObserverEntry>} */ records
+      ) =>
+        records.forEach(({target, isIntersecting}) => {
+          this.getEntry_(target).updateVisibility(
+            /* isVisible */ isIntersecting
+          );
+        });
+      this.viewportObserver_ = getViewportObserver(
+        viewportCallback,
+        this.ampdoc.win
+      );
+    }
+    this.viewportObserver_.observe(
+      /** @type {!AMP.BaseElement} */ (video).element
+    );
 
     this.entries_ = this.entries_ || [];
     const entry = new VideoEntry(this, video);
-    this.maybeInstallVisibilityObserver_(entry);
     this.entries_.push(entry);
+    listen(
```

Comment:
Ditto. If we listen, should also unlisten in `unregister()`, which we don't have :)

Reply:
The previous code wasn't unlistening either :O. There isn't a great place for it :/. There is a `dispose` function but I don't think its ever used. Also this is a Service not a component, so theres no natural unlayout.

Reply:
register() is called from build, which doesn't have a corollary unbuild()

Reply:
If so, no prob. We can supply the unbuild callback. Here's my recommendation: let's add the `unregister()` method, but proceed without calling it for now. I'll send a pull request for the paired build/unbuild and we can then finish it all up once we are happy about it.

Reply:
I'll add a TODO to create a corresponding `unregister()`, since its also unclear to me if we have a `removeAction` at all